include/eve/module/complex/complex.hpp

```diff
@@ -251,6 +251,59 @@ namespace eve
       return y * x;
     }
 
+    template<like<complex> Z1, like<complex> Z2>
+    friend auto
+    tagged_dispatch(eve::tag::mul_, pedantic_type const &, Z1 const& z1, Z2  const& z2) noexcept
```

Comment:
Are you sure that pedantic mul should me in the main file?

Reply:
Have you another proposal ?

Reply:
up to now, in complex the decorated version of the functions are in the same file as the functions, but I can put them in details arithmetic. Not a great difference.
Tell me your pro and con

Reply:
I think maybe our design principles changed here maybe.
Originally we wanted to do "include what you use" model, so that user can save compile time.
Meaning, in this case  the would need to include the header for pedantic multiplication.

Now we have these big headers that we want users to include, assuming that modules will be here soon. So now this is less impactful.

never the less, I'd probably go like we do everywhere else:

`regular/mul` `pedantic/mul` and then include one of both of them, something like that.

Reply:
all the way the use of complex is to include all complex functions with no details. Talk with Joel.

Reply:
We changed when we went for the modular approach. Include time of functions is mostly the same this way. 
For complex, we need a lot of them anyway and we wanted to mimic std so includign eve/complex.hpp gives you everything proper out of the box.